src/main/java/fr/openent/pmb/services/impl/DefaultUserService.java

```diff
@@ -0,0 +1,71 @@
+package fr.openent.pmb.services.impl;
+
+import fr.openent.pmb.services.UserService;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.entcore.common.neo4j.Neo4j;
+import org.entcore.common.neo4j.Neo4jResult;
+
+import static io.vertx.core.http.impl.HttpClientConnection.log;
+
+public class DefaultUserService implements UserService {
+
+    public Future<JsonArray> getUsersByStructure(JsonArray UAIs, String type){
+        Promise<JsonArray> promise = Promise.promise();
+        String query = "MATCH (s:Structure)<-[:DEPENDS]-(cpg:ProfileGroup)"+
+                "OPTIONAL MATCH (cpg)-[:HAS_PROFILE]->(p:Profile)\n" +
+                "OPTIONAL MATCH (cpg)<-[:IN]-(u:User)\n" +
+                "WHERE s.UAI IN {uais}\n" +
+                "AND (p.name IN {type})\n" +
+                "RETURN DISTINCT\n" +
+                "u.externalId as externalId,\n" +
+                "u.email as email,\n" +
+                "u.address as address,\n" +
+                "u.zipCode as zipCode,\n" +
+                "u.country as country,\n" +
+                "u.city as city;";
+        JsonObject params = new JsonObject()
+                .put("uais", UAIs)
+                .put("type", type);
+
+        Neo4j.getInstance().execute(query, params, Neo4jResult.validResultHandler(event -> {
+            if (event.isLeft()) {
+                log.error("Pmb-Connector@DefaultUserService::getUsersByStructure Error when trying to execute query");
```

Comment:
met des [ ] pour que ce soit + iso à toutes les plateformes

Reply:
Par ex startPromise.tryFail("[Competences@Competences::start] Fail to start Competences");